### PR TITLE
Log rotation - possibility of log rotations of 2,3,4,5,x minutes or hours

### PR DESCRIPTION
**Pull request by leopucci**
After: Log name based on file change
Before: Log name based on rotation timestamp.

This patch solves the problem of filenames containing 00-00-00 and adds the possibility of log rotations of 2,3,4,5 m or hours (not only 1m,1h)
Important to mention that log rotation is based on logging.metrics.period. So if you put a time of 10s but logging.metrics.period default is 30s, it will rotate logs each 30s.

This patch solves https://github.com/elastic/beats/issues/11365

### CODE REVIEW

**Comment by leopucci** [comment]
@jsoriano this will change the behavior of all interval logs from all beats, could someone review it?

**Comment by jsoriano** [comment]
@leopucci thanks for investigating this issue! There are some tests failing in CI, could you take a look?

**Comment by leopucci** [comment]
@jsoriano Thanks for the heads up.. 

TestYearlyRotator
    interval_rotator_test.go:209: 
        	Error Trace:	interval_rotator_test.go:209
        	Error:      	Not equal: 
        	            	expected: "foo-2019-"
        	            	actual  : "foo-2019-03-"

**Comment by leopucci** [comment]
@jsoriano Done! :) \o/

**Comment by leopucci** [comment]
Ci will fail on different values, but now seems that all tests are ok!

File: libbeat/common/file/rotator.go [hunk 1]
```diff
@@ -378,14 +378,15 @@ func (r *Rotator) rotate(reason rotateReason) error {
...
+	_, err := os.Stat(r.filename)
+
 	if os.IsNotExist(err) {
 		return nil
 	} else if err != nil {
 		return errors.Wrap(err, "failed to rotate backups")
 	}
 
-	logPrefix := r.intervalRotator.LogPrefix(r.filename, fi.ModTime())
+	logPrefix := r.intervalRotator.LogPrefix(r.filename, time.Now())
```
Comment (michalpristas) [inline libbeat/common/file/rotator.go]: what about the situation where last entry was yesterday then service was down and now we're writing again?
shouldn't log say that the file we're trying to backup was created and logged into yesterday only?

Comment (leopucci) [reply libbeat/common/file/rotator.go]: IIt is a matter of taste now.... 
What is more important on log rotation? See the exactly time that log rotated or see the last time the file was written?
Inside the log I can see dates and times..

Comment (leopucci) [reply libbeat/common/file/rotator.go]: Check this out... Log rotated with file mod date time. The service went up and generated a file that I don't know if rotated in the past or when the service started again.it is better to know when the service started again?

54924387-08987100-4eeb-11e9-920c-850ec3b8660d

**Comment by leopucci** [comment]
@michalpristas  I am going to return the behavior of file renaming, although i don't agree, i think this needs more discussion. 
For now i will just solve the bug.

File: libbeat/common/file/rotator.go (hunk 1 above)
Comment (leopucci) [reply libbeat/common/file/rotator.go]:
```suggestion
	logPrefix := r.intervalRotator.LogPrefix(r.filename, fi.ModTime())
```

**Comment by michalpristas** [comment]
hey @leopucci thanks for the PR, 
I will take a look and try to play with it a bit to see how this behaves today or on Monday 

can you also make those tests pass?

**Comment by leopucci** [comment]
@michalpristas I will analyze them to see how to make tests pass :)

**Comment by leopucci** [comment]
@michalpristas I have correted the tests. 

There is a difference that before, you had fixed times and for all other time values an arbitrary time function. 
No you can use 1,2,3,4,5 sec/min/hour/etc to rotate the time. As a consequence of being allowed to choose time, you do not have arbitrary time anymore. So I removed the arbitrary tests and the code of it. 
Please take a look

**Comment by leopucci** [comment]
I will move this mod to a branch of my fork in some minutes .. stand by

**Comment by leopucci** [comment]
@michalpristas  I'm closing this pull request. The correct one is https://github.com/elastic/beats/pull/11602

**Pull request closed without merging**